source/virtualBuffers/MSHTML.py

```diff
@@ -44,6 +44,10 @@ def _normalizeFormatField(self, attrs):
 		language=attrs.get('language')
 		if language:
 			attrs['language']=languageHandler.normalizeLanguage(language)
+		textPosition = attrs.get('textPosition')
+		if textPosition and textPosition not in ['super', 'sub']:
+			del attrs['text-position']
+		log.debug(f'{attrs}')

```

Comment:
Would you consider making text position an Enum, and extracting this to a separate method so it works in the same way as `def _getIsCurrentAttribute`  (line 53)?

Then type checking can ensure that the value that shows up in the speech module is only one of the expected values. Someone wishing to add a new value in the future can more easily follow the enum type through the code.

Reply:
Do you mean using a new `controlTypes.TextPosition` enum for all the text-position format fields passed from the various APIs to `getFormatFieldSpeech`?
If yes:
* would you recommend to define this enum with only 2 values (sub and super)? This would lead to a code a bit strange mixing up the enum values with None in some cases (baseline or other value or missing value).
* On the opposit, we could have an enum with 3 values (sub, super and baseline). But it does not follow anymore the suggestion made by @michaelDCurran in https://github.com/nvaccess/nvda/pull/11897#issuecomment-813802247.
In any case, the code of all the APIs would need to be rework to normalize the values to the enum's one. It is actually better than today's code since an unexpected value coming from an API is checked closer to the source.

Reply:
Yes, I personally think that three explicit values makes more sense. Note, that two strings and `None` is also three known values, they just don't share a consistent type.

For each value there should be a mapping to the speech for that value, the enum `BASE_LINE` value should have an empty string (or maybe `None`) for it's speech, take a look at how `isCurrent` works.

Reply:
This is still unresolved @CyrilleB79.

Reply:
> This is still unresolved @CyrilleB79.

Yes, I need to work on this but I need to take a little time to see and check all the cases.
Putting this PR in draft for now.